tests/gold/deepArrayCopy.c

```diff
@@ -0,0 +1,128 @@
+#include "deepArrayCopy.h"
+
+
+struct awl_awl_unsignedS32 * copyArrayPos_awl_awl_unsignedS32(struct awl_awl_unsignedS32 * dst, int32_t dstLen, struct awl_awl_unsignedS32 * src, int32_t srcLen, int32_t pos)
+{
+  for (int32_t i = 0; i < srcLen; i += 1)
+  {
+    (dst[(pos + i)]).buffer = initCopyArray_awl_unsignedS32((dst[(pos + i)]).buffer, (dst[(pos + i)]).length, (src[i]).buffer, (src[i]).length);
+    (dst[(pos + i)]).length = (src[i]).length;
+  }
+  return(dst);
+}
+
+struct awl_awl_unsignedS32 * copyArray_awl_awl_unsignedS32(struct awl_awl_unsignedS32 * dst, int32_t dstLen, struct awl_awl_unsignedS32 * src, int32_t srcLen)
+{
+  dst = copyArrayPos_awl_awl_unsignedS32(dst, dstLen, src, srcLen, 0);
+  return(dst);
+}
+
+struct awl_awl_unsignedS32 * initCopyArray_awl_awl_unsignedS32(struct awl_awl_unsignedS32 * dst, int32_t dstLen, struct awl_awl_unsignedS32 * src, int32_t srcLen)
+{
+  dst = initArray_awl_awl_unsignedS32(dst, dstLen, srcLen);
+  dstLen = srcLen;
+  dst = copyArrayPos_awl_awl_unsignedS32(dst, dstLen, src, srcLen, 0);
+  return(dst);
+}
+
+struct awl_unsignedS32 * copyArrayPos_awl_unsignedS32(struct awl_unsignedS32 * dst, int32_t dstLen, struct awl_unsignedS32 * src, int32_t srcLen, int32_t pos)
+{
+  for (int32_t i = 0; i < srcLen; i += 1)
+  {
+    (dst[(pos + i)]).buffer = initCopyArray((dst[(pos + i)]).buffer, (dst[(pos + i)]).length, sizeof(uint32_t), (src[i]).buffer, (src[i]).length);
+    (dst[(pos + i)]).length = (src[i]).length;
+  }
+  return(dst);
+}
+
+struct awl_unsignedS32 * copyArray_awl_unsignedS32(struct awl_unsignedS32 * dst, int32_t dstLen, struct awl_unsignedS32 * src, int32_t srcLen)
+{
+  dst = copyArrayPos_awl_unsignedS32(dst, dstLen, src, srcLen, 0);
+  return(dst);
+}
+
+struct awl_unsignedS32 * initCopyArray_awl_unsignedS32(struct awl_unsignedS32 * dst, int32_t dstLen, struct awl_unsignedS32 * src, int32_t srcLen)
+{
+  dst = initArray_awl_unsignedS32(dst, dstLen, srcLen);
+  dstLen = srcLen;
+  dst = copyArrayPos_awl_unsignedS32(dst, dstLen, src, srcLen, 0);
```

Comment:
(Unrelated to this change). Why not do the copy propagation manually and call copyArrayPos with srcLen for both length arguments?